charts/argo-cd/templates/dex/deployment.yaml

```diff
@@ -99,8 +99,8 @@ spec:
         {{- if .Values.dex.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /healthz/live
-            port: metrics
+            path: {{ .Values.dex.livenessProbe.httpPort }}
+            port: {{ .Values.dex.livenessProbe.httpPath }}
```

Comment:
The port name is hardcoded in deployment. This cannot be configured.

Reply:
Same applies most likely to path. Configuration is automatically generated by init container here.
Ref: https://github.com/argoproj/argo-cd/blob/4dc91dcb9df77071d0d8a83511449e2297d984b1/util/dex/config.go#L44C10-L44C19

Reply:
I understand your concern. This is not an ideal solution but for users relying on Google Cloud OIDC, the solution proposed in https://github.com/argoproj/argo-cd/issues/9091 seems to be a viable way to ensure dex server is restarted when it is not able to open the google connector. 

The default values are not introducing any behaviour changes so only chart users specifically changing the proposed values would see a difference.

The change would help, I understand some users may compromise their deployment if they configure a wrong value here, but if I can provide some documentation of the possible values or with a helm template function to ensure the set port is in the list of existing ports

WDYT ?

Reply:
Hi, thanks for providing context about this change. I think it's ok to allow configuration for this edge case.